### PR TITLE
Use scipy's axis-angle computation for rel. error computation until we patch GTSAM

### PR DESCRIPTION
**Pull request by johnwlambert**
Use scipy's axis-angle computation for rel. error computation until we patch GTSAM

See https://github.com/borglab/gtsam/issues/886 for more details.

Unblocks us for now in GTSFM.

### CODE REVIEW

File: tests/utils/test_geometry_comparisons.py [hunk 1]
```diff
@@ -221,6 +222,41 @@ def test_compute_relative_rotation_angle(self):
 
         np.testing.assert_allclose(computed_deg, expected_deg, rtol=1e-3, atol=1e-3)
 
+
+    def test_compute_relative_rotation_angle(self) -> None:
```
Comment (ayushbaid) [inline tests/utils/test_geometry_comparisons.py:226]: How does this test pass when GTSAM's comparisons are buggy?

File: gtsfm/utils/geometry_comparisons.py [hunk 2]
```diff
@@ -275,7 +275,11 @@ def compute_relative_rotation_angle(R_1: Optional[Rot3], R_2: Optional[Rot3]) ->
         return None
 
     relative_rot = R_1.between(R_2)
-    relative_rot_angle_rad = relative_rot.axisAngle()[1]
+    # TODO(johnwlambert): we are using scipy.spatial.transform as a hotfix until GTSAM axisAngle() is patched.
+    # See https://github.com/borglab/gtsam/issues/886
+    scaled_axis = Rotation.from_matrix(relative_rot.matrix()).as_rotvec()
+    relative_rot_angle_rad = np.linalg.norm(scaled_axis)
+    axis = scaled_axis / relative_rot_angle_rad
```
Comment (ayushbaid) [inline gtsfm/utils/geometry_comparisons.py:282]: Do we need to cast to scipy rotation? Can we get the rotation vector from gtsam?

File: tests/utils/test_geometry_comparisons.py (hunk 1 above)
Comment (johnwlambert) [reply tests/utils/test_geometry_comparisons.py:226]: @ayushbaid So the GTSAM axisAngle *usually* works. Just not for a few edge cases. Over 50,000 image pairs, this can be 100+ completely erroneous values, making our plots look like garbage. Scipy seems to handle these edge cases properly currently. So this unit test here is checking some generic cases, but can’t sample the edge cases. So it passes.

File: gtsfm/utils/geometry_comparisons.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/utils/geometry_comparisons.py:282]: @ayushbaid hmm this is what we were doing before, right? But it was buggy.

**Review by ayushbaid — APPROVED** [review]

**Comment by ProfFan** [comment]
@johnwlambert  Please test if the fix works and maybe revert this?
